app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Any
+
+
+def cache(func: callable) -> callable:
+    cache_vault = {}
+
+    def inner(*args) -> Any:
+        if args in cache_vault:
+            print("Getting from cache")
+            return cache_vault[args]
+        print("Calculating new result")
+        result = func(*args)
+        cache_vault[args] = result
+        return result
```

Comment:
You are returning the same value, try using only one return.

Reply:
Updated...